contest_runner.py

```diff
@@ -319,64 +319,79 @@ def upload_file(file_full_path, remote_name=None, remove_local=False):
-            "max_steps": self.max_steps,
-            "organizer": self.organizer,
-            "timestamp_id": self.contest_timestamp_id,
+            'games': self.games,
+            'team_stats': self.team_stats,
+            'random_layouts': [l for l in self.layouts if l.startswith('RANDOM')],
+            'fixed_layouts': [l for l in self.layouts if not l.startswith('RANDOM')],
+            'max_steps': self.max_steps,
+            'organizer' : self.organizer,
+            'timestamp_id' : self.contest_timestamp_id
         }
 
-        # Process replays: compress and upload
-        replays_archive_name = "replays_%s.tar" % self.contest_timestamp_id
-        replays_archive_name += ".gz" if self.compress_logs else ""
-        replays_archive_full_path = os.path.join(
-            self.replays_archive_dir, replays_archive_name
-        )
-        with tarfile.open(
-            replays_archive_full_path, "w:gz" if self.compress_logs else "w"
-        ) as tar:
-            tar.add(self.tmp_replays_dir, arcname="/")
+        # PROCESS REPLAYS: compress and upload
+        replays_archive_name = 'replays_%s.tar' % self.contest_timestamp_id
+        replays_archive_name += '.gz' if self.compress_logs else ''
+        replays_archive_full_path = os.path.join(self.replays_archive_dir, replays_archive_name)
+        with tarfile.open(replays_archive_full_path, 'w:gz' if self.compress_logs else 'w') as tar:
+            tar.add(self.TMP_REPLAYS_DIR, arcname='/')
         if self.upload_replays:
             try:
-                replays_file_url = self.upload_file(
-                    replays_archive_full_path, remove_local=False
-                )
-                data_stats["url_replays"] = replays_file_url.decode()
+                replays_file_url = self.upload_file(replays_archive_full_path, remove_local=False)
+                data_stats['url_replays'] = replays_file_url.decode()
             except Exception as e:
-                replays_file_url = os.path.relpath(
-                    replays_archive_full_path, self.www_dir
-                )
+                replays_file_url = os.path.relpath(replays_archive_full_path, self.www_dir)
         else:
-            replays_file_url = os.path.relpath(
-                replays_archive_full_path, self.www_dir
-            )  # stats-archive/stats_xxx.json
-
-        # Process replays: compress and upload
-        logs_archive_name = "logs_%s.tar" % self.contest_timestamp_id
-        logs_archive_name += ".gz" if self.compress_logs else ""
+            replays_file_url = os.path.relpath(replays_archive_full_path, self.www_dir)  # stats-archive/stats_xxx.json
+
+        # Copy folder
+        replays_folder_name = 'replays_%s' % self.contest_timestamp_id
+        replays_archive_full_path = os.path.join(self.replays_archive_dir, replays_folder_name)
+        shutil.copytree(self.TMP_REPLAYS_DIR, replays_archive_full_path)
```

Comment:
Here check because I don't think we use `self.TMP_REPLAYS_DIR` anymore. 

I think it is `self.tmp_replays_dir` as it is not a constant

Reply:
Yes this was true, fixed when merged from master!